src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,122 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEAFULT_SIZE = 16;
+    static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    int size;
+
+    private Entry<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Entry[DEAFULT_SIZE];
+        size = 0;
+    }
+
+    private static class Entry<K, V> {
+        K key;
+        V value;
+        Entry<K, V> next;
+
+        Entry(K key, V value, Entry<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
+        int index = getIndex(key);
+        Entry<K, V> current = table[index];
+
+        if (key == null) {
+            if (current == null) {
+                table[index] = new Entry<>(key, value, null);
```

Comment:
Use of raw types is discouraged. Replace 'new Entry<>(...)' with 'new Entry<K, V>(...)' to ensure type safety.